services/outputhost/cheramiCommitter.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package outputhost
+
+import (
+	"sync"
+	"time"
+
+	"github.com/uber/cherami-server/common"
+	"github.com/uber/cherami-thrift/.generated/go/metadata"
+	"github.com/uber/tchannel-go/thrift"
+)
+
+const metaContextTimeout = 10 * time.Second
+
+// cheramiCommitter is commits ackLevels to Cassandra through the TChanMetadataClient interface
+type cheramiCommitter struct {
+	outputHostUUID     string
+	cgUUID             string
+	extUUID            string
+	connectedStoreUUID *string
+	commitLevel        CommitterLevel
+	readLevel          CommitterLevel
+	finalLevel         CommitterLevel
+	metaclient         metadata.TChanMetadataService
+	m                  sync.Mutex
+}
+
+/*
+ * Committer interface
+ */
+
+// Commit just updates the next Cherami ack level that will be flushed
+func (c *cheramiCommitter) Commit(l CommitterLevel) {
+	c.m.Lock()
```

Comment:
As we discussed this might a bit too heavyweight since we do it for every message. We can have a locker for the committer which takes it once during some operation and drops it once it is done..